File: app/src/main/java/com/sketch/ragnarok/BlurFilter.java
package com.sketch.ragnarok;

import android.graphics.Bitmap;

class BlurFilter {
    static {
        System.loadLibrary("AndroidImageFilter");
    }

    static Bitmap changeToAverageBlur(Bitmap bitmap, int maskSize) {
        if (maskSize % 2 == 0) {
            throw new IllegalArgumentException(String.format("the maskSize must odd, but %d is an even", maskSize));
        }
        int width = bitmap.getWidth();
        int height = bitmap.getHeight();

        int[] pixels = new int[width * height];
        bitmap.getPixels(pixels, 0, width, 0, 0, width, height);

        int[] returnPixels = NativeFilterFunc.averageSmooth(pixels, width, height, maskSize);

        return Bitmap.createBitmap(returnPixels, width, height, Bitmap.Config.ARGB_8888);
    }
}

File: app/src/main/java/com/sketch/ragnarok/BlurFilter.kt
package com.sketch.ragnarok

import android.graphics.Bitmap

internal object BlurFilter {

    init {
        System.loadLibrary("AndroidImageFilter")
    }

    fun changeToAverageBlur(bitmap: Bitmap, maskSize: Int): Bitmap {
        require(maskSize % 2 != 0) {
            String.format(
                "the maskSize must odd, but %d is an even", maskSize
            )
        }
        val width = bitmap.width
        val height = bitmap.height
        val pixels = IntArray(width * height)
        bitmap.getPixels(pixels, 0, width, 0, 0, width, height)
        val returnPixels = NativeFilterFunc.averageSmooth(pixels, width, height, maskSize)
        return Bitmap.createBitmap(returnPixels, width, height, Bitmap.Config.ARGB_8888)
    }
}